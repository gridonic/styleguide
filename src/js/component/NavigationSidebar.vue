<template>
    <div class="navigation-sidebar -typography-styleguide">
        <div
            v-for="section in sections"
            :key="section.title"
            class="section"
            :data-section="section.title"
        >
            <h4>{{ section.title }}</h4>
            <a
                v-for="item in section.items"
                :key="item.title"
                :href="'#' + item.title"
                class="link"
            >
                {{ item.title }}
            </a>
        </div>
    </div>
</template>

<script>
export default {
    name: 'NavigationSidebar',
    props: {
        components: {
            type: Array,
            default: () => [],
        },
    },
    computed: {
        sections() {
            return [
                {
                    title: 'Basic',
                },
                {
                    title: 'Components',
                    items: this.components.map(c => ({
                        title: c.name,
                    })),
                },
            ];
        },
    },
};
</script>

<style lang="scss" scoped>
    .navigation-sidebar {
        padding: 10px;

        > .section {
            margin-bottom: 10px;

            > .link {
                display: block;
                padding-left: 10px;
                margin-top: 5px;
                margin-bottom: 5px;
            }
        }
    }
</style>
