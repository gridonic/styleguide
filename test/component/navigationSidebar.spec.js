import { shallowMount } from '@vue/test-utils';
import NavigationSidebar from '@/js/component/NavigationSidebar';

describe('given main sidebar component', () => {
    function build() {
        const wrapper = shallowMount(NavigationSidebar);

        return {
            wrapper,
            componentSection: () => wrapper.find('[data-section="Components"]'),
        };
    }

    it('displays when rendered', () => {
        const { wrapper } = build();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('displays section titles', () => {
        const { wrapper } = build();

        const sections = wrapper.findAll('.section');
        expect(sections.length > 0).toBeTruthy();
    });

    it('contains at least basic and components sections', () => {
        const { wrapper } = build();

        expect(wrapper.find('[data-section="Basic"]').exists())
            .toBeTruthy();

        expect(wrapper.find('[data-section="Components"]').exists())
            .toBeTruthy();
    });

    describe('given no components', () => {
        it('contains no component links', () => {
            const { componentSection } = build();

            expect(componentSection().findAll('.component').length)
                .toBe(0);
        });
    });

    describe('given components', () => {
        function buildWithComponents() {
            const { wrapper, componentSection } = build();

            wrapper.setProps({
                components: [
                    { name: 'Primary Button' },
                    { name: 'Search Box' },
                ],
            });

            return {
                wrapper,
                componentSection,
                sectionLinks: section => section().findAll('.link'),
            };
        }

        it('contains one link for each component', () => {
            const { componentSection, sectionLinks } = buildWithComponents();

            expect(sectionLinks(componentSection).length)
                .toBe(2);
        });

        it('component links href="#name-of-compoment"', () => {
            const { componentSection, sectionLinks } = buildWithComponents();

            const links = sectionLinks(componentSection);

            expect(links.at(0).attributes('href'))
                .toEqual('#Primary Button');

            expect(links.at(1).attributes('href'))
                .toEqual('#Search Box');
        });
    });
});
