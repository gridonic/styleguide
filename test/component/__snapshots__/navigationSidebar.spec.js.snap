// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given main sidebar component displays when rendered 1`] = `"<div class=\\"navigation-sidebar -typography-styleguide\\"><div data-section=\\"Basic\\" class=\\"section\\"><h4>Basic</h4> </div><div data-section=\\"Components\\" class=\\"section\\"><h4>Components</h4> </div></div>"`;
